internal/services/authorization/role_management_policy_resource.go

```diff
@@ -477,9 +461,9 @@ func (r RoleManagementPolicyResource) Read() sdk.ResourceFunc {
 									state.ActivationRules[0].RequireApproval = pointer.From(settings.IsApprovalRequired)
 									if settings.ApprovalStages != nil {
 										approvalStages := *settings.ApprovalStages
-										state.ActivationRules[0].ApprovalStages = make([]RoleManagementPolicyApprovalStage, 1)
 										approvalStage := approvalStages[0]
 										if primaryApprovers := approvalStage.PrimaryApprovers; primaryApprovers != nil && len(*primaryApprovers) > 0 {
+											state.ActivationRules[0].ApprovalStages = make([]RoleManagementPolicyApprovalStage, 1)
```

Comment:
Only make a new approval stage in state once we are sure there is one in Azure. Otherwise, all plans thereafter will indicate that activations_rules.approval_stage needs to be changed to nil.